pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTestBase.java

```diff
@@ -127,29 +128,44 @@ protected void addDataToH2(Map<String, List<GenericRow>> rowsMap)
   public void setUp()
       throws Exception {
     DataTableBuilderFactory.setDataTableVersion(DataTableFactory.VERSION_4);
-    QueryServerEnclosure server1 = new QueryServerEnclosure(
-        ImmutableMap.of("a", INDEX_DIR_S1_A, "b", INDEX_DIR_S1_B, "c", INDEX_DIR_S1_C, "d_O", INDEX_DIR_S1_D),
-        QueryEnvironmentTestUtils.SERVER1_SEGMENTS);
-    QueryServerEnclosure server2 = new QueryServerEnclosure(
-        ImmutableMap.of("a", INDEX_DIR_S2_A, "c", INDEX_DIR_S2_C, "d_R", INDEX_DIR_S2_D, "d_O", INDEX_DIR_S1_D),
-        QueryEnvironmentTestUtils.SERVER2_SEGMENTS);
+    MockInstanceDataManagerFactory factory1 = new MockInstanceDataManagerFactory("server1")
+        .registerTable(QueryEnvironmentTestBase.SCHEMA_BUILDER.setSchemaName("a").build(), "a_REALTIME")
```

Comment:
tables must have segments, right? should we consider making them one API call? `registertable(schema, name, segment...)`

Reply:
table can have 0 segments